matlab/bart.m

```diff
@@ -1,27 +1,42 @@
 function [varargout] = bart(cmd, varargin)
-% BART	Call BART command from Matlab.
-%   [A, B] = bart('command', X, Y) call command with inputs X Y and outputs A B
+% BART Call BART command from Matlab.
+%   [varargout] = BART(cmd, varargin) to run given bart command (cmd) using the
+%   data arrays/matrices passed as varargin.
 %
+%   [A, B] = BART('command', X, Y) call command with inputs X Y and outputs A B
+%
+%   To output a list of available bart commands simply run "bart". To
+%   output the help for a specific bart command type "bart command -h".
+%
+% Parameters:
+%   cmd:        Command to run as string (including non data parameters)
+%   varargin:   Data arrays/matrices used as input
+%
+% Example:
+%   bart traj -h
+%   [reco] = bart('nuft -i traj', data) call nufft with inputs data and outputs reco
+%
+% Authors:
 % 2014-2016 Martin Uecker <uecker@med.uni-goettingen.de>
-% Edited for WSL, by Soumick Chatterjee <soumick.chatterjee@ovgu.de>
-
+% 2018 (Edited for WSL) Soumick Chatterjee <soumick.chatterjee@ovgu.de>
+% 2020 Martin Krämer <martin.kraemer@med.uni-jena.de>
 
+    % Check input variables
 	if nargin==0 || all(cmd==0)
-		disp('Usage: bart <command> <arguments...>');
-		return
+		fprintf('Usage: bart <command> <arguments...>\n\n');
+		cmd = '';
 	end
 
+    % Check bart toolbox path
 	bart_path = getenv('TOOLBOX_PATH');
 	isWSL = false;
-
 	if isempty(bart_path)
 		if exist('/usr/local/bin/bart', 'file')
 			bart_path = '/usr/local/bin';
 		elseif exist('/usr/bin/bart', 'file')
 			bart_path = '/usr/bin';
 		else
-			% Try to execute bart inside wsl, if it works, then it returns
-			% status 0
+			% Try to execute bart inside wsl, if it works, then it returns status 0
 			[bartstatus, ~] = system('wsl bart version -V');
 			if bartstatus==0
 				bart_path = '/usr/bin';
```

Comment:
As far as I can see, the variable bart_path is actually not used for the WSL code branch below.

Reply:
That is correct, I'm running bart on a native ubuntu, not using wsl - so I did not bother checking. Nevertheless, I changed the code to use bart_path as well in the wsl part, see 9e988d46ab861ead321d8e06aa10a4a2a28beb3

Reply:
OK, but how do we know that bart is in /usr/bin ?

Reply:
We don't, as far as I understand, this probably very ancient piece of code, it just assumes this as default location in case the TOOLBOX_PATH is not set. Because WSL 2 is supposed to be released soon ([https://docs.microsoft.com/en-us/windows/wsl/wsl2-index](https://docs.microsoft.com/en-us/windows/wsl/wsl2-index) this will probably require an overhaul with WSL 2 anyway.

Reply:
OK, got it. I might play with it once I upgrade to WSL 2.